### PR TITLE
ONS Styling Changeover

### PR DESCRIPTION
**Pull request by matthewandrewpalmer**
- Changing Styling to use the new ONS Style Guide.
- Header formatting and removal of validation item 
- Made a copy of the bootstrap dropdown and model styling to be able to remove the rest of bootstrap styling.
- Made temp ONS table styling css file so that all tables look right before being altered.
- Some buttons may be back but will be changed when the individual page is changed.
- Created new Form Field macro to reuse the fields on every page, setup on the login page and new/edit user page for now for testing.

image

### CODE REVIEW

File: blaise_survey_manager/static/css/ONS-Table.css [hunk 1]
```diff
@@ -0,0 +1,189 @@
+table {
```
Comment (cadmiumcat) [inline blaise_survey_manager/static/css/ONS-Table.css]: Is there a reason not to use the cdn instead?

Comment (cadmiumcat) [reply blaise_survey_manager/static/css/ONS-Table.css]: It had previously been added in a folder as we couldn't access the cdn from on-prem

File: blaise_survey_manager/users/templates/upload_csv.html [hunk 2]
```diff
@@ -1,75 +1,62 @@
...
-                                {{ field(class_="field__item input input--text input-type__input js-typeahead-input input--upload form-control is-invalid", autocomplete="new-password") }}
-                                {% for error in field.errors %}
-                                    <div class="invalid-feedback">{{ error }}</div>
-                                {% endfor %}
-                            {% else %}
-                                {{ field(class_="field__item input input--text input-type__input js-typeahead-input input--upload form-control", autocomplete="new-password") }}
-                            {% endif %}
+    <div class="page__container container" id="content">
+        <main id="main-content" class="page__main ">
+            <p><a style="margin-bottom: 20px" href="{{ url_for('users_view.list_users') }}">Previous</a></p>
```
Comment (cadmiumcat) [inline blaise_survey_manager/users/templates/upload_csv.html]: keep styling ouf of here, I think

File: blaise_survey_manager/static/css/ONS-Table.css (hunk 1 above)
Comment (matthewandrewpalmer) [reply blaise_survey_manager/static/css/ONS-Table.css]: This was done as the ONS styled tables need the styling to be set by the class, this just applies the ONS styling based on the table tags, s they tables don't look blank. I'm changing the styling for the tables when I'm improving each page.

File: blaise_survey_manager/auth/templates/login.html [hunk 3]
```diff
@@ -1,39 +1,29 @@
...
-        </div>
+    {{ super() }}
+    <div class="page__container container" id="content">
+        <main id="main-content" class="page__main ">
+            <h1>Login</h1>
+            <form action="" method="post" class="form" role="form">
+                {{ form.csrf_token }}
+                {{ render_text_field(form.username, true) }}
+                {{ render_password_field(form.password) }}
+                <br>
```
[the remark this answers is not in the document]
Comment (matthewandrewpalmer) [reply blaise_survey_manager/auth/templates/login.html]: Started replacing these for most pages, excusing any allocation pages. And the survey config pages as i've already altered the styling for that.

File: blaise_survey_manager/templates/base.html [hunk 4]
```diff
@@ -5,9 +5,25 @@
 {% endblock html_attribs %}
 
 {% block head %}
-    {{ super() }}
-	<link rel="stylesheet" href="/static/responsive.css">
+{#    {{ super() }}#}
+    <title>{% block title %}{{title|default}}{% endblock title %}</title>
+    <meta name="msapplication-config" content="https://cdn.ons.gov.uk/sdc/design-system/15.0.0/favicons/browserconfig.json">
```
Comment (cadmiumcat) [inline blaise_survey_manager/templates/base.html]: So glad to have cdn back 👍

**Comment by matthewandrewpalmer** [comment]
> Should "Alert" and "Warning" checks be together is the category is set to the same thing ? e.g. Error, or should there be a "warning" category ? (this is in blaise_survey_manager/templates/base.html)

I was looking at the messages that are set as Warnings and most of then are about errors and failures, I i added them in the same category so they have the right status, before I change the pages.

**Pull request merged**